src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.h

```diff
@@ -0,0 +1,160 @@
+class SpinningBodyTwoDOFStateEffector: public StateEffector, public SysModel {
+public:
+    double mass1 = 0.0;                 //!< [kg] mass of lower spinning body (can be 0)
+    double mass2 = 1.0;                 //!< [kg] mass of upper spinning body
+    double k1 = 0.0;                    //!< [N-m/rad] torsional spring constant for first rotation axis
+    double k2 = 0.0;                    //!< [N-m/rad] torsional spring constant for second rotation axis
+    double c1 = 0.0;                    //!< [N-m-s/rad] rotational damping coefficient for first rotation axis
+    double c2 = 0.0;                    //!< [N-m-s/rad] rotational damping coefficient for second rotation axis
+    double theta1Init = 0.0;            //!< [rad] initial first axis angle
+    double theta1DotInit = 0.0;         //!< [rad/s] initial first axis angle rate
+    double theta2Init = 0.0;            //!< [rad] initial second axis angle
+    double theta2DotInit = 0.0;         //!< [rad/s] initial second axis angle rate
+    std::string nameOfTheta1State;      //!< -- identifier for the theta1 state data container
+    std::string nameOfTheta1DotState;   //!< -- identifier for the thetaDot1 state data container
+    std::string nameOfTheta2State;      //!< -- identifier for the theta2 state data container
+    std::string nameOfTheta2DotState;   //!< -- identifier for the thetaDot2 state data container
+    Eigen::Vector3d r_S1B_B;            //!< [m] vector pointing from body frame B origin to lower spinning frame S1 origin in B frame components
+    Eigen::Vector3d r_S2S1_S1;          //!< [m] vector pointing from lower spinning frame S1 origin to upper spinning frame S2 origin in S1 frame components
+    Eigen::Vector3d r_Sc1S1_S1;         //!< [m] vector pointing from lower spinning frame S1 origin to point Sc1 (center of mass of the lower spinner) in S1 frame components
+    Eigen::Vector3d r_Sc2S2_S2;         //!< [m] vector pointing from upper spinning frame S2 origin to point Sc2 (center of mass of the upper spinner) in S2 frame components
+    Eigen::Vector3d s1Hat_S1;           //!< -- first spinning axis in S1 frame components.
+    Eigen::Vector3d s2Hat_S2;           //!< -- second spinning axis in S2 frame components.
+    Eigen::Matrix3d IS1PntSc1_S1;       //!< [kg-m^2] Inertia of lower spinning body about point Sc1 in S1 frame components
+    Eigen::Matrix3d IS2PntSc2_S2;       //!< [kg-m^2] Inertia of upper spinning body about point Sc2 in S2 frame components
+    Eigen::Matrix3d dcm_S10B;           //!< -- DCM from the body frame to the S10 frame (S1 frame for theta1=0)
+    Eigen::Matrix3d dcm_S20S1;          //!< -- DCM from the S1 frame to the S20 frame (S2 frame for theta2=0)
+    std::vector<Message<HingedRigidBodyMsgPayload>*> spinningBodyOutMsgs;       //!< vector of state output messages
+    std::vector<Message<SCStatesMsgPayload>*> spinningBodyConfigLogOutMsgs;     //!< vector of spinning body state config log messages
```

Comment:
:large_blue_circle: Why is `Log` in this variable name? It seems out of place. The message is just output. What happens to it afterwards is up to the user. They may log it, they may throw it away, they may connect it to another module.

Reply:
I agree with you, but this is the way all effectors name their output kinematic state messages. I will keep it for consistency, we can do a refactor of all effectors down the line.